src/vision/status_quo/alan_needs_async_in_traits.md

```diff
@@ -0,0 +1,101 @@
+  |
+  = note: `async` trait functions are not currently supported
+  = note: consider using the `async-trait` crate: https://crates.io/crates/async-trait
+```
+
+Alan, using Rust for a little while now, has learned to follow compiler error messages and adds `async-trait` to the dependencies list. Alan follows the README of `async-trait` and gets the code to compile. Alan is somewhat dismayed that the compile times have increased considerably after adding this new dependency.
+
+**TODO::** Maybe add representive pre and post trait compile times here.
+
+After Alan finishes adding the new trait, he shows his work off to Barbara and mentions he's happy with the work but is a little sad that compile times have worsened. Barbara, an experienced Rust developer, knows that using `async-trait` comes with some issues and in this case is especially worried about tying their public API to a third-party dependency. She decides to not worry Alan with this right now. Alan and Barbara are pretty happy with the results and go on to publish their crate which gets lots of users.
+
+Later on, a potential user of the library wants to use their library in a `no_std` context where they will be providing a custom HTTP stack. Alan and Barbara have done a pretty good job of limiting the use of standard library features and think it might be possible to support this use case. However, they quickly run into a show stopper: `async-trait` boxes all of the futures returned from a async trait function. They report this to Alan through an issue.
+
+Alan, feeling (over-) confident in his Rust skills, decides to try to see if he can implement async traits without using `async-trait`. 
+
+```rust 
+struct Request(String);
+
+trait HttpClient {
+   type Response: Future<Output = ()>;
+
+   fn perform(request: Request) -> Self::Response; 
+}
+```
+
+Alan seems to have something working, but when he goes to update the examples of how to implement this trait in his crate's documentation, he realizes that he either needs to:
+
+* use trait object:
+
+  ```rust,ignore
+  struct ClientImpl;
+
+  impl HttpClient for ClientImpl {
+      type Response = Pin<Box<dyn Future<Output = ()>>>;
+
+      fn perform(request: Request) -> Self::Response {
+          Box::pin(async move {
+              println!("{}", request.0);
+          })
+	  }
+  }
+  ```
+
+  which wouldn't work for `no_std`.
+
+* implement `Future` trait manually, which isn't particulary easy/straight-forward for non-trivial cases, especially if it involves making other async calls (likely).
+
+**TODO::** Would make sense for Alan to reach out to Barbara at this point and she should be able to give some good advice.
+
+## 🤔 Frequently Asked Questions
+
+### **What are the morals of the story?**
+
+* `async-trait` is awesome, but has some drawbacks
+    * compile time increases
+    * performance cost of boxing and dynamic dispatch
+    * lack of control over whether futures are `Send` or `Sync`
```

Comment:
`async-trait` *does* give you control over this. You can use `#[async_trait(?Send)]` if you don't want to add `Send` bounds.

Reply:
I'd say we should remove this bullet